bigbluebutton-html5/imports/ui/components/components-data/chat-context/adapter.jsx

```diff
@@ -62,10 +62,14 @@ const Adapter = () => {
   ChatLogger.trace('chatAdapter::body::users', users[Auth.meetingID]);
 
   useEffect(() => {
-    window.addEventListener(EVENT_NAME, () => {
-      dispatch({
-        type: ACTIONS.CLEAR_STREAM_MESSAGES,
-      });
+    window.addEventListener(EVENT_NAME, (e) => {
+      const { role } = e.detail;
+
+      if (role !== usersData[Auth.userID].role) {
```

Comment:
What's the role impact in messages cleaning?

Reply:
#12012 added the messages cleaning, but we only needed to clean messages if the user role changes (#11842)

Reply:
Taking a little more attention, it looks like there may be a race condition, as you are using to compare the role from user collection and role from user context. I think the information may take a little longer to update to the context, but there is no 100% guarantee that this logic will work every time.

Reply:
moved role check to `chat-context/adapter` in 1a032e4